templates/profile/haproxy/service.cfg.erb

```diff
@@ -46,26 +46,46 @@ EOT
     "  server #{hostname} #{ip}:#{port} #{check} cookie s#{ip.split('.').last}"
   end
 
-  def any_whitelists?
-    @whitelists.find { |_,exemptions| exemptions.size() > 0 }
+  def nonempty_whitelists
+    @whitelists.select { |_,exemptions| exemptions.size() > 0 }
   end
 
-  def whitelist_acl_files
-    @whitelists.select { |_,exemptions| exemptions.size() > 0 }
-      .map do |whitelist,_|
+  def any_exemptions?
+    nonempty_whitelists.size() > 0 or @throttle_condition
+  end
+
+  def whitelist_file_acls
+    nonempty_whitelists.map do |whitelist,_|
       "  acl whitelist_#{whitelist} #{whitelist} -n -f /etc/haproxy/#{@service}_whitelist_#{whitelist}.txt\n"
-      end.join("")
+    end.join("")
   end
 
-  def whitelist_condition
-    @whitelists.select { |_,exemptions| exemptions.size() > 0 }
-      .map do |whitelist,_|
-        "whitelist_#{whitelist}"
-      end.join(" OR ")
+  def throttle_condition_acl
+    if @throttle_condition
+      "  acl throttle_condition #{@throttle_condition}\n"
+    else
+      ""
+    end
   end
 
-  def whitelist(service_prefix)
-    whitelist_acl_files + "  use_backend #{service_prefix}-back-exempt if #{whitelist_condition}\n"
+  def whitelist_acl_names
+    nonempty_whitelists.map do |whitelist,_|
+      "whitelist_#{whitelist}"
+    end
+  end
+
+  def exemption_condition
+    acls = whitelist_acl_names
+    acls = acls + ["!throttle_condition"] if @throttle_condition
+    acls.join(" OR ")
+  end
+
+  def exemption(service_prefix)
+    if any_exemptions?
+      "  use_backend #{service_prefix}-back-exempt if #{exemption_condition}\n"
+    else
+      ""
+    end
```

Comment:
Part of me always feels like this is exactly how not to use erb files, but I honestly kind of love this. Everything you need is just right there, but there's still a clear separation between logic and format.

Anyway "Let's Get The Manual"